dask/dataframe/io/csv.py

```diff
@@ -68,6 +70,9 @@ def pandas_read_text(reader, b, header, kwargs, dtypes=None, columns=None,
         raise ValueError("Columns do not match", df.columns, columns)
     elif columns:
         df.columns = columns
+    if path:
+        df = df.assign(**{path[0]: path[1]})
```

Comment:
You would avoid building an array of objects, parsing for uniques and making the categorical by doing directly
```
df.assign(**{path[0]: pd.Categorical.from_codes(np.zeros(len(df)), [path[1]])})
```
(yes, I know that looks ugly!)

Reply:
That's find if that is the recommended approach. Does that work for all versions of pandas?

Reply:
`from_codes` has been around a while. @TomAugspurger  ?

Reply:
If we want `known` categories (which I think we do), all filenames should be passed to all calls to `read_text` to be added to their categories. So the `path` arg above would also need to include that information, and this would look more like:

```python
colname, path, paths = path
code = paths.index(path)
df.assign(**{colname: pd.Categorical.from_codes(np.full(len(df), code), paths)})
```

> Is it better to use a list rather than a numpy array? I just noticed that there are no other references to numpy in the file.

Pandas requires numpy, and imports it internally - there's no harm in importing it here.